File: app/src/androidTest/java/art/manguste/android/ArasakaWeapons/MainActivityBasicTest.java
package art.manguste.android.ArasakaWeapons;

import androidx.test.espresso.intent.Intents;
import androidx.test.rule.ActivityTestRule;
import androidx.test.runner.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.intent.Intents.intended;
import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
import static androidx.test.espresso.matcher.ViewMatchers.withId;


@RunWith(AndroidJUnit4.class)
public class MainActivityBasicTest {

    @Rule
    public ActivityTestRule<MainActivity> mActivityTestRule =
            new ActivityTestRule<>(MainActivity.class);

    @Test
    public void clickCartButton_GoToOrderActivity() {
        Intents.init();
        // click on cart button
        onView(withId(R.id.cartImage)).perform(click());

        // check activity opens
        intended(hasComponent(OrderActivity.class.getName()));
        Intents.release();
    }
}

File: app/src/androidTest/java/art/manguste/android/arasakaWeapons/MainActivityBasicTest.kt
package art.manguste.android.arasakaWeapons

import androidx.test.espresso.Espresso
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.intent.Intents
import androidx.test.espresso.intent.matcher.IntentMatchers
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.rule.ActivityTestRule
import androidx.test.runner.AndroidJUnit4
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MainActivityBasicTest {
    @Rule
    var mActivityTestRule = ActivityTestRule(MainActivity::class.java)
    @Test
    fun clickCartButton_GoToOrderActivity() {
        Intents.init()
        // click on cart button
        Espresso.onView(ViewMatchers.withId(R.id.cartImage)).perform(ViewActions.click())

        // check activity opens
        Intents.intended(IntentMatchers.hasComponent(OrderActivity::class.java.name))
        Intents.release()
    }
}